### PR TITLE
Fix polynomial with slope 0

### PR DESCRIPTION
**Pull request by jonathanlew**
### Description
Fixed a bug where the `i` was of type `size_t` (which is unsigned) and was used in a `for` loop where the exit condition required `i` to be negative. Instead of crashing, this bug caused `getOrder` to loop forever. 

I've made this mistake before, so it's a case where better code conventions can prevent this: #1402

### Testing Done
added a regresstion test

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [ ] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [ ] **Remove all commented out code**
- [ ] **Remove extra print statements**: for example, those just used for testing
- [ ] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [ ] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

File: src/software/new_geom/polynomial1d.cpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ unsigned int Polynomial1d::getOrder() const
 {
     if (coeffs.size() != 0)
     {
-        for (size_t i = coeffs.size() - 1; i >= 0; i--)
+        for (int i = coeffs.size() - 1; i >= 0; i--)
```
Comment (garethellis0) [inline src/software/new_geom/polynomial1d.cpp]: why not just iterate in the other direction? I suppose it's likely that we're going to need to iterate over a few more elements, but I think the overall performance hit seems pretty minimal?

**Review by garethellis0 — CHANGES_REQUESTED** [review]

**Review by EvanMorcom — DISMISSED** [review]
LGTM 🚜 🌆

File: src/software/new_geom/polynomial1d.cpp (hunk 1 above)
Comment (garethellis0) [reply src/software/new_geom/polynomial1d.cpp]: Another option is to offset `i` by 1 so you can iterate until `i > 0` instead. 

Also we noticed during in-person discussion that we're doing a signed - unsigned comparison here.

Comment (jonathanlew) [reply src/software/new_geom/polynomial1d.cpp]: I did the offset i by 1 thing. it's way more clean

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1:

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Review by EvanMorcom — APPROVED** [review]
LGTM 🔥

**Pull request merged**